hazelcast/src/hazelcast/client/spi/InvocationService.cpp

```diff
@@ -267,7 +275,17 @@ namespace hazelcast {
                 }
 
                 boost::shared_ptr<connection::CallPromise> promise = deRegisterCall(connection, correlationId);
-                if (promise.get() == NULL) {
+                if (NULL == promise.get()) {
```

Comment:
If I don't remember wrong, it  is possible to get null from deregister in some cases . But I need to focus on that more to figure it out. Can you reconsider this part if there could be another thread already deregistered in some cases(exception, connection close .. etc) ?